### DISCUSSION TITLE
A collection of troublesome LAS files for test cases

### DISCUSSION THREAD

**Discussion by EvanBianco** [kinverarity1/lasio#436] [Show and tell]
I've started collecting a list of LAS files that I've encountered that won't load using lasio for one reason or another:

https://github.com/agile-geoscience/LASinfections

Some issues might be solvable with some clever engineering, other cases might benefit from some better error handling, the simplest of which might be, "Please go an fix your data". I haven't been able to identify the reason the file fails to load in all cases, and some of these examples might have multiple issues.

Please feel free to add examples to this, use these files elsewhere for test cases, or discuss as you wish.

**Comment by kinverarity1** [comment]
Thank you for these! I appreciate seeing "wild" LAS files as I don't see them much myself.

There was a keyword argument added to a recent release a few weeks ago which will probably help to load some of these (haven't looked through in detail yet):

```python
>>> las = lasio.read('example.las', ignore_header_errors=True)
```

It will replace LASHeaderError exceptions with logger.warnings, and just drop the offending line(s). At least until the parser can be improved to read in those lines. Still not in the documentation, which is something to be done.
